services/slam/server.go

```diff
@@ -149,15 +150,76 @@ func (server *subtypeServer) GetInternalState(ctx context.Context, req *pb.GetIn
 func (server *subtypeServer) GetPointCloudMapStream(req *pb.GetPointCloudMapStreamRequest,
 	stream pb.SLAMService_GetPointCloudMapStreamServer,
 ) error {
-	return errors.New("unimplemented stub")
+	ctx := context.Background()
+
+	ctx, span := trace.StartSpan(ctx, "slam::server::GetPointCloudMapStream")
+	defer span.End()
+
+	svc, err := server.service(req.Name)
+	if err != nil {
+		return err
+	}
+
+	f, err := svc.GetPointCloudMapStream(ctx, req.Name)
+	if err != nil {
+		return errors.Wrap(err, "getting callback function from GetPointCloudMapStream encountered an issue")
+	}
+
+	// Channel buffer can be used here to optimize for latency
+	for {
+		rawChunk, err := f()
+
+		if errors.Is(err, io.EOF) {
+			return nil
```

Comment:
Why don't we want to pass along the EOF, so that the caller can handle it? Or are we supposed to close `stream` when we get EOF?

Reply:
yeah, we just close the stream when we see the EOF. When the stream closes, the server will send an EOF error to the client.

Reply:
Do we need to do something to close the stream here?

Reply:
as far as we can tell from testing and research, no.

Reply:
Ah yes, I see this in the docs as well: https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc